app/src/main/java/com/example/tasky/common/data/SessionStateManagerManagerImpl.kt

```diff
@@ -13,6 +15,16 @@ class SessionStateManagerManagerImpl @Inject constructor(
         private const val TAG = "SessionStateManager"
     }
 
+    override suspend fun setAccessToken(accessToken: String) {
+        appPreferences.put(PreferencesKeys.ACCESS_TOKEN, accessToken)
+    }
+
+    override fun getAccessToken(): String? {
+        return runBlocking {
+            appPreferences.get(PreferencesKeys.ACCESS_TOKEN).firstOrNull()
+        }
```

Comment:
Avoid using runBlocking here when you have no certainty that this is not called from the UI thread. Instead just expose a suspend function like above and execute this on a "real" scope (usually viewModelScope)